forage-android/src/main/java/com/joinforage/forage/android/network/data/BaseVaultRequestParams.kt

```diff
@@ -0,0 +1,16 @@
+package com.joinforage.forage.android.network.data
+
+internal open class BaseVaultRequestParams(
+    open val cardNumberToken: String,
+    open val encryptionKey: String
+) {
+    override fun equals(other: Any?): Boolean {
+        return other is BaseVaultRequestParams &&
+            other.cardNumberToken == cardNumberToken &&
+            other.encryptionKey == encryptionKey
+    }
+
+    override fun hashCode(): Int {
+        return cardNumberToken.hashCode() + encryptionKey.hashCode()
+    }
```

Comment:
`data classes` do not support inheritance.

The overridden `equals` and `hashcode` are needed when `BaseVaultRequestParams` is used inside a `Map` (like in our unit tests, and possible other places in the future).

Open to learning about alternative workarounds.